pkg/csi_driver/node.go

```diff
@@ -93,8 +98,13 @@ func (s *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 
 		// Login credentials
 
-		options = append(options, s.driver.config.Username)
-		options = append(options, s.driver.config.Password)
+		secrets := req.GetNodePublishSecrets()
+		if err := validateNodePublishSecrets(secrets); err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+
+		options = append(options, secrets[smbUser])
+		options = append(options, secrets[smbPassword])
```

Comment:
Can we enhance https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/pkg/csi_driver/node_test.go#L60 to verify that the mount options are correct?

Reply:
@almos98 is this done now?